src/AspNet.Security.OAuth.Salesforce/SalesforceAuthenticationHandler.cs

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using AspNet.Security.OAuth.Extensions;
+using JetBrains.Annotations;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.OAuth;
+using Microsoft.AspNetCore.Http.Authentication;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.Salesforce
+{
+    public class SalesforceAuthenticationHandler : OAuthHandler<SalesforceAuthenticationOptions>
+    {
+        public SalesforceAuthenticationHandler([NotNull] HttpClient client)
+            : base(client)
+        {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens)
+        {
+            var identityServiceUrl = tokens.Response.Value<string>("id");
+            var salesforceInstanceUrl = tokens.Response.Value<string>("instance_url");
```

Comment:
So the "userinfo" endpoint is not static (which explains why it doesn't have a default value in the options) and must be resolved at runtime?